### PR TITLE
SASS-9021: All 9021 changes

### PR DESCRIPTION
**Pull request by ghousedev**
### Description
Move to backend mongo session, behind fs.  New branch to fix PR.

##### Before Reviewing
- [ ]  Have you pulled the branch down?
- [ ]  Have you assigned yourself to the PR?
- [ ]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [ ]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [ ]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [ ]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [ ]  Have you addressed warnings where appropriate?
- [ ]  Have you rebased against the current version of main?
- [ ]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

**Review by Shihabur-Atos — DISMISSED** [review]
LGTM

File: app/controllers/CreateDividendsSessionDataController.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+class CreateDividendsSessionDataController @Inject()(dividendsSessionService: DividendsSessionService,
+                                                     cc: ControllerComponents,
+                                                     authorisedAction: AuthorisedAction)
+                                                    (implicit ec: ExecutionContext) extends BackendController(cc) {
+
+  def createSessionData(taxYear: Int): Action[AnyContent] = authorisedAction.async { implicit user =>
+    user.body.asJson.map(_.validate[DividendsCheckYourAnswersModel]) match {
+      case Some(JsSuccess(model, _)) =>
+        dividendsSessionService.createSessionData(model, taxYear)(NotModified)(NoContent) // review NotModified error.. should be internalServerError
```
Comment (ansahdavid) [inline app/controllers/CreateDividendsSessionDataController.scala:37]: Can this comment be removed

File: app/controllers/CreateStockDividendsSessionDataController.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+class CreateStockDividendsSessionDataController @Inject()(stockDividendsSessionService: StockDividendsSessionService,
+                                                          cc: ControllerComponents,
+                                                          authorisedAction: AuthorisedAction)
+                                                         (implicit ec: ExecutionContext) extends BackendController(cc) {
+
+  def createSessionData(taxYear: Int): Action[AnyContent] = authorisedAction.async { implicit user =>
+    user.body.asJson.map(_.validate[StockDividendsCheckYourAnswersModel]) match {
+      case Some(JsSuccess(model, _)) =>
+        stockDividendsSessionService.createSessionData(model, taxYear)(NotModified)(NoContent) // review NotModified error.. should be internalServerError
```
Comment (ansahdavid) [inline app/controllers/CreateStockDividendsSessionDataController.scala:37]: Can this comment be removed

File: app/controllers/DeleteDividendsIncomeDataController.scala [hunk 3]
```diff
@@ -28,10 +28,10 @@ class DeleteDividendsIncomeDataController @Inject()(deleteDividendsIncomeDataSer
...
 
-  def deleteDividendsIncomeData(nino: String, taxYear: Int): Action[AnyContent] = authorisedAction.async {
-    implicit user =>
+  def deleteDividendsIncomeData(nino: String, taxYear: Int): Action[AnyContent] = authorisedAction.async { implicit user =>
       deleteDividendsIncomeDataService.deleteDividendsIncomeData(nino, taxYear).map {
-        case Right(true) => NoContent
-        case Left(errorModel) => Status(errorModel.status)(errorModel.toJson)      }
+        case Right(_) => NoContent
+        case Left(errorModel) => Status(errorModel.status)(errorModel.toJson)
+      }
```
Comment (ansahdavid) [inline app/controllers/DeleteDividendsIncomeDataController.scala:35]: Can we add a line / space to satisfy the Scalastyle formatting

File: app/controllers/UpdateDividendsSessionDataController.scala [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+class UpdateDividendsSessionDataController @Inject()(dividendsSessionService: DividendsSessionService,
+                                                     cc: ControllerComponents,
+                                                     authorisedAction: AuthorisedAction)
+                                                    (implicit ec: ExecutionContext) extends BackendController(cc) {
+
+  def updateSessionData(taxYear: Int): Action[AnyContent] = authorisedAction.async { implicit user =>
+    user.body.asJson.map(_.validate[DividendsCheckYourAnswersModel]) match {
+      case Some(JsSuccess(model, _)) =>
+        dividendsSessionService.updateSessionData(model, taxYear)(NotModified)(NoContent) // review NotModified error.. should be internalServerError
```
Comment (ansahdavid) [inline app/controllers/UpdateDividendsSessionDataController.scala:37]: Can this comment be removed

File: app/controllers/UpdateStockDividendsSessionDataController.scala [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+class UpdateStockDividendsSessionDataController @Inject()(stockDividendsSessionService: StockDividendsSessionService,
+                                                          cc: ControllerComponents,
+                                                          authorisedAction: AuthorisedAction)
+                                                         (implicit ec: ExecutionContext) extends BackendController(cc) {
+
+  def updateSessionData(taxYear: Int): Action[AnyContent] = authorisedAction.async { implicit user =>
+    user.body.asJson.map(_.validate[StockDividendsCheckYourAnswersModel]) match {
+      case Some(JsSuccess(model, _)) =>
+        stockDividendsSessionService.updateSessionData(model, taxYear)(NotModified)(NoContent) // review NotModified error.. should be internalServerError
```
Comment (ansahdavid) [inline app/controllers/UpdateStockDividendsSessionDataController.scala:37]: Can this comment be removed

File: app/models/RefreshIncomeSourceRequest.scala [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+package models
+
+import play.api.libs.json.{Json, OFormat}
+
+case class RefreshIncomeSourceRequest(incomeSource: String)
+
+object RefreshIncomeSourceRequest {
+  implicit val formats: OFormat[RefreshIncomeSourceRequest] = Json.format[RefreshIncomeSourceRequest]
+}
```
Comment (ansahdavid) [inline app/models/RefreshIncomeSourceRequest.scala:25]: Can we add a line / space to satisfy the Scalastyle formatting

File: app/models/User.scala [hunk 7]
```diff
@@ -18,6 +18,9 @@ package models
 
 import play.api.mvc.{Request, WrappedRequest}
 
-case class User[T](mtditid: String, arn: Option[String])(implicit val request: Request[T]) extends WrappedRequest[T](request) {
+//TODO: investigate the use of this model and values, possibly remove at least affinityGroup
```
Comment (ansahdavid) [inline app/models/User.scala:21]: I think this comment be removed

File: app/models/dividends/StockDividendsSubmissionModel.scala [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+                                         dividendIncomeReceivedWhilstAbroad: Option[Seq[ForeignDividendModel]],
+                                         stockDividend: Option[StockDividendModel],
+                                         redeemableShares: Option[StockDividendModel],
+                                         bonusIssuesOfSecurities: Option[StockDividendModel],
+                                         closeCompanyLoansWrittenOff: Option[StockDividendModel])
+
+object StockDividendsSubmissionModel {
+
+  implicit val formats: OFormat[StockDividendsSubmissionModel] = Json.format[StockDividendsSubmissionModel]
+}
```
Comment (ansahdavid) [inline app/models/dividends/StockDividendsSubmissionModel.scala:31]: And here

File: project/plugins.sbt [hunk 9]
```diff
@@ -3,6 +3,6 @@ resolvers += Resolver.url("HMRC-open-artefacts-ivy", url("https://open.artefacts
 
 addSbtPlugin("uk.gov.hmrc"        % "sbt-auto-build"          % "3.22.0")
 addSbtPlugin("uk.gov.hmrc"        % "sbt-distributables"      % "2.5.0")
-addSbtPlugin("org.playframework"  % "sbt-plugin"              % "3.0.2")
+addSbtPlugin("org.playframework"  % "sbt-plugin"              % "3.0.3")
 addSbtPlugin("org.scoverage"      % "sbt-scoverage"           % "2.0.9")
 addSbtPlugin("org.scalastyle"     %% "scalastyle-sbt-plugin"  % "1.0.0" exclude("org.scala-lang.modules", "scala-xml_2.12"))
```
Comment (ansahdavid) [inline project/plugins.sbt:8]: And here

File: test/connectors/httpParsers/APIParserSpec.scala [hunk 10]
```diff
@@ -50,22 +49,24 @@ class APIParserSpec extends TestUtils{
...
     }
+
     "handle single errors" in {
       val result = FakeParser.handleAPIError(httpResponse(Json.parse(
         """{"code":"INTERNAL_SERVER_ERROR","reason":"The service is currently facing issues."}""".stripMargin)))
       result mustBe Left(ErrorModel(INTERNAL_SERVER_ERROR,ErrorBodyModel("INTERNAL_SERVER_ERROR","The service is currently facing issues.")))
     }
   }
-
 }
```
Comment (ansahdavid) [inline test/connectors/httpParsers/APIParserSpec.scala:72]: Can we add a line / space to satisfy the Scalastyle formatting

File: test/support/stubs/AppConfigStub.scala [hunk 11]
```diff
@@ -38,5 +40,15 @@ class AppConfigStub extends MockFactory {
...
+    override val encryptionKey: String = "1234556"
+
+    override def mongoTTL: Long = Duration("28").toDays
+
+    override val tailoringEnabled: Boolean = true
+    override val interestTailoringEnabled: Boolean = true
+    override val dividendsTailoringEnabled: Boolean = true
+    override val incomeTaxSubmissionBEBaseUrl: String = "http://localhost:9308"
   }
 }
```
Comment (ansahdavid) [inline test/support/stubs/AppConfigStub.scala:54]: Can we add a line / space to satisfy the Scalastyle formatting

**Review by ansahdavid — APPROVED** [review]

**Review by Shihabur-Atos — APPROVED** [review]
LGTM

**Review by ansahdavid — APPROVED** [review]

**Review by Shihabur-Atos — APPROVED** [review]
LGTM

**Pull request merged**